include/cxx.h

```diff
@@ -828,89 +828,241 @@ Box<T>::Box(uninit) noexcept {}
 
 #ifndef CXXBRIDGE1_RUST_OPTION
 template <typename T>
-class Option final {
+class Option final {};
+
+template <typename T>
+class Option<T&> {
 public:
   Option() noexcept;
   Option(Option&&) noexcept;
-  Option(T&&) noexcept;
+  Option(T&) noexcept;
   ~Option() noexcept;
 
-  const T *operator->() const;
-  const T &operator*() const;
-  T *operator->();
-  T &operator*();
-
-  Option<T>& operator=(Option&&) noexcept;
+  Option<T&>& operator=(Option&&) noexcept;
 
   bool has_value() const noexcept;
   T& value() noexcept;
   void reset();
-  void set(T&&) noexcept;
+  void set(T&) noexcept;
+  static Option<T&> from_raw(T*) noexcept;
+  T* into_raw() noexcept;
+private:
+  void* repr;
+
+  void drop() noexcept;
+};
+
+template <typename T>
+class Option<const T&> {
+public:
+  Option() noexcept;
+  Option(const Option&) noexcept;
```

Comment:
So `Option<const T&>` is copyable, but `Option<T&>` is not? Just checking that this was intentional

Reply:
So if you copy an Option<T&>, you would now have 2 mutable references to the same value which I think would be instant UB... Option<Box<T>> is not copy for similar reasons.